### PR TITLE
tests: unset LD_PRELOAD while executing match script

### PR DESCRIPTION
**Pull request by GBuella**
During some libpmemfile tests, the library was preloaded
while executing perl. This can result in hard to understand
build errors, when libpmemfile happens to be very buggy, e.g. perl
not being able to load perl modules:

```
Can't locate Getopt/Std.pm in @INC (you may need to install the Getopt::Std module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.22.1 /usr/local/share/perl/5.22.1 /usr/lib/x86_64-linux-gnu/perl5/5.22 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.22 /usr/share/perl/5.22 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base .) at /pmemfile/build-deb/debug/pmemfile/tests/match line 74.
BEGIN failed--compilation aborted at /pmemfile/build-deb/debug/pmemfile/tests/match line 74.
```

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/143)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/143#-:-KkVyheh2BkpFwg-4Klj:bnfp4nl)*

File: tests/helpers.cmake [hunk 1]
```diff
@@ -43,30 +43,48 @@ function(common_cleanup)
...
+	if(PRELOAD_LIB)
+		unset(ENV{LD_PRELOAD})
+	endif()
 	execute_process(COMMAND ${MKFS_EXECUTABLE} ${path} ${size}
 			RESULT_VARIABLE HAD_ERROR)
 	if(HAD_ERROR)
 		message(FATAL_ERROR "mkfs(${path}, ${size}) failed: ${HAD_ERROR}")
 	endif()
+	if(PRELOAD_LIB)
+		set(ENV{LD_PRELOAD} ${PRELOAD_LIB})
```
Comment (marcinslusarz) [inline tests/helpers.cmake]: Please restore previous value of LD_PRELOAD.

Comment (GBuella) [inline tests/helpers.cmake]: Done

**Comment by krzycz** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/143)*

**Comment by marcinslusarz** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/143#-:-KkkZ6-I9F8PtbuJ_DBU:bnfp4nl)*

**Pull request merged**